plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/LoginCredentialExporter.java

```diff
@@ -0,0 +1,77 @@
+package com.google.cloud.tools.eclipse.appengine.login.ui;
+
+import java.io.IOException;
+import java.nio.file.Files;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.jface.window.IShellProvider;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.cloud.tools.eclipse.appengine.login.CredentialHelper;
+import com.google.cloud.tools.eclipse.appengine.login.GoogleLoginService;
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Charsets;
+
+/**
+ * Provides helper methods to save and retrieve an OAuth refresh token obtained via using Google OAuth service
+ *
+ * @see GoogleLoginService
+ * @see CredentialHelper
+ */
+public class LoginCredentialExporter {
+
+  private static final String CREDENTIAL_FILENAME = "gcloud-credentials.json";
+  
+  private GoogleLoginService loginService;
+  private CredentialHelper credentialHelper;
+  
+  public LoginCredentialExporter() {
+    this(new GoogleLoginService(), new CredentialHelper());
+  }
+  
+  @VisibleForTesting
+  LoginCredentialExporter(GoogleLoginService googleLoginService, CredentialHelper credentialHelper) {
+    this.loginService = googleLoginService;
+    this.credentialHelper = credentialHelper;
+  }
+
+  /**
+   * Retrieves the OAuth credentials via {@link GoogleLoginService#getActiveCredential(IShellProvider)} (which may
+   * initiate a web-based login process if there is no active logged in user present) and saves the credentials to
+   * a json file under <code>workDirectory</code>
+   * <p>
+   * <b>This method must be called from the UI thread.</b>
+   *
+   * @param workDirectory the destination for the json file containing the received OAuth credential
+   * @param shellProvider used if a web-based logn process is needed
+   * @throws IOException if communication with the login service fails or the json file with the credential
+   * cannot be created
+   * @throws CoreException if the login attempt was unsuccessful
+   */
+  public void logInAndSaveCredential(IPath workDirectory, IShellProvider shellProvider) throws IOException, 
```

Comment:
It seems strange for a `LoginCredentialExporter` to initiate UI and affect login state.